lib/state.js

```diff
@@ -308,36 +323,82 @@ StateResolver.prototype.resolveLocation_ = function(frame) {
   };
 };
 
-StateResolver.prototype.resolveArgumentList_ = function(frame) {
+StateResolver.prototype.extractArgumentsList_ = function (frame) {
   var args = [];
   for (var i = 0; i < frame.argumentCount(); i++) {
     // Don't resolve unnamed arguments.
     if (!frame.argumentName(i)) {
       continue;
     }
-    args.push(this.resolveVariable_(
-        frame.argumentName(i), frame.argumentValue(i)));
+    args.push({name: frame.argumentName(i), value: frame.argumentValue(i)});
   }
   return args;
 };
 
-StateResolver.prototype.resolveLocalsList_ = function(frame,
-    resolvedArguments) {
-  var locals = [];
-  // Arguments may have been captured as locals in a nested closure.
-  // We filter them out here.
-  var predicate = function(localEntry, argEntry) {
-    return argEntry.varTableIndex === localEntry.varTableIndex;
-  };
-  for (var i = 0; i < frame.localCount(); i++) {
-    var localEntry = this.resolveVariable_(
-        frame.localName(i), frame.localValue(i));
-    if (!resolvedArguments.some(predicate.bind(null, localEntry))) {
-      locals.push(this.resolveVariable_(
-        frame.localName(i), frame.localValue(i)));
+StateResolver.prototype.resolveArgumentList_ = function(args) {
+  var resolveVariable = this.resolveVariable_.bind(this);
+  return args.map(function (arg){
+    return resolveVariable(arg.name, arg.value);
+  });
+};
+
+/**
+ * Iterates and returns variable information for all scopes (excluding global)
+ * in a given frame. FrameMirrors should return their scope object list with 
+ * most deeply nested scope first so variables initially encountered will take
+ * precedence over subsequent instance with the same name - this is tracked in
+ * the usedNames map. The argument list given to this function may be
+ * manipulated if variables with a deeper scope occur which have the same name.
+ * @function resolveLocalsList_
+ * @memberof StateResolver
+ * @param {FrameMirror} frame - A instance of FrameMirror
+ * @param {Array<Object>} args - An array of objects representing any function
+ *  arguments the frame may list
+ * @returns {Array<Object>} - returns an array containing data about selected
+ *  variables
+ */
+StateResolver.prototype.resolveLocalsList_ = function (frame, args) {
+  var usedNames = {};
+  var makeMirror = this.ctx_.MakeMirror;
+  // resolveVariable_ accesses instance properties - bind to instance
+  var resolveVariable = this.resolveVariable_.bind(this);
+  return flatten(frame.allScopes().map(
+    function (scope) {
+      switch (scope.scopeType()) {
```

Comment:
In the constructor, or at top level you can run:
```js
var ScopeType = require('vm').runInDebugContext('ScopeType')
```
and then use `ScopeType.Global` instead of hard-coding the constants here.